packages/js/src/plugins/auctionHouseModule/operations/cancelBid.ts

```diff
@@ -105,28 +118,42 @@ export const cancelBidBuilder = (
 ): TransactionBuilder<CancelBidBuilderContext> => {
   const { auctionHouse, auctioneerAuthority, bid } = params;
 
-  if (auctionHouse.hasAuctioneer && !auctioneerAuthority) {
+  // Data.
+  const {
+    asset,
+    buyerAddress,
+    tradeStateAddress,
+    price,
+    receiptAddress,
+    tokens,
+    isPublic,
+  } = bid;
+  const {
+    authorityAddress,
+    address: auctionHouseAddress,
+    feeAccountAddress,
+    hasAuctioneer,
+  } = auctionHouse;
+
+  if (hasAuctioneer && !auctioneerAuthority) {
     throw new AuctioneerAuthorityRequiredError();
   }
 
-  // Data.
-  const { asset, tradeStateAddress, price, tokens, isPublic } = bid;
-
   // Accounts.
   const tokenAccount = isPublic
     ? findAssociatedTokenAccountPda(
         asset.mint.address,
-        toPublicKey(bid.buyerAddress)
+        toPublicKey(buyerAddress)
       )
-    : bid.asset.token.address;
+    : (asset as SftWithToken | NftWithToken).token.address;
```

Comment:
Why do we need the type hint here?

Reply:
TypeScript doesn't see typing that comes from isPublic, so It doesn't treat assets as SftWithToken | NftWithToken by default

Reply:
Thanks!